server/plugin_test.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+import (
+	"errors"
+	"io/ioutil"
+	"testing"
+
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/mattermost/mattermost-server/plugin/plugintest"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+func mockReader(filename string) ([]byte, error) {
+	return []byte{0}, nil
+}
+
+func TestSetupBot(t *testing.T) {
+	t.Run("EnsureBot error", func(t *testing.T) {
+		plugin := &Plugin{}
+		helpers := &plugintest.Helpers{}
+		helpers.On("EnsureBot", mock.Anything).Return("", errors.New(""))
+		plugin.SetHelpers(helpers)
+		defer helpers.AssertExpectations(t)
+		err := plugin.setupBot(nil)
+		assert.NotNil(t, err)
+	})
+
+	t.Run("GetBundlePath error", func(t *testing.T) {
+		plugin := &Plugin{}
+		helpers := &plugintest.Helpers{}
+		helpers.On("EnsureBot", mock.Anything).Return("bot", nil)
+		plugin.SetHelpers(helpers)
+		defer helpers.AssertExpectations(t)
+		api := &plugintest.API{}
+		api.On("GetBundlePath").Return("", errors.New(""))
+		plugin.SetAPI(api)
+		defer api.AssertExpectations(t)
+		err := plugin.setupBot(nil)
+		assert.NotNil(t, err)
+	})
+
+	t.Run("wrong file path", func(t *testing.T) {
+		plugin := &Plugin{}
+		helpers := &plugintest.Helpers{}
+		helpers.On("EnsureBot", mock.Anything).Return("bot", nil)
+		plugin.SetHelpers(helpers)
+		defer helpers.AssertExpectations(t)
+		api := &plugintest.API{}
+		api.On("GetBundlePath").Return("some path", nil)
+		plugin.SetAPI(api)
+		defer api.AssertExpectations(t)
+		err := plugin.setupBot(ioutil.ReadFile)
+		assert.NotNil(t, err)
+	})
+
+	t.Run("SetProfileImage error", func(t *testing.T) {
+		plugin := &Plugin{}
+		helpers := &plugintest.Helpers{}
+		helpers.On("EnsureBot", mock.Anything).Return("bot", nil)
+		plugin.SetHelpers(helpers)
+		defer helpers.AssertExpectations(t)
+		api := &plugintest.API{}
+		api.On("GetBundlePath").Return("some path", nil)
+		api.On("SetProfileImage", mock.Anything, mock.Anything).Return(model.NewAppError("", "", nil, "", 404))
+		plugin.SetAPI(api)
+		defer api.AssertExpectations(t)
+		err := plugin.setupBot(mockReader)
```

Comment:
It's not worth to test this just for the code coverage IMO. `plugin.setupBot` is not a public method and hence should not be tested.